### PR TITLE
Can pass function to plugins array [closes #60]

### LINKED ISSUE #60: Ability to pass function to plugins array for autoprefixer-stylus

### PR DESCRIPTION
**Pull request by artisin**
- [x] tests
- [x] documented
- [x] bumped package

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -76,7 +76,9 @@ class StylusCompiler {
     if (Array.isArray(plugins)) {
       const handler = plugin => compiler.use(plugin());
       plugins.forEach(pluginName => {
-        if (Array.isArray(pluginName)) {
+        if (Object.prototype.toString.call(pluginName) === '[object Function]') {
```
Comment (shvaikalesh) [inline index.js]: `type pluginName === 'function'`

**Comment by shvaikalesh** [comment]
Hey @artisin, thanks for the PR. Currently, latest versions of Brunch have issues with compiler chaining. We will fix that soon. `autoprefixer-{stylus,sass,less}` is only a nasty workaround.

However, I like the change, please change function detection and remove version bump and I will merge this.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "stylus-brunch",
-  "version": "2.8.0",
+  "version": "2.8.1",
```
Comment (shvaikalesh) [inline package.json]: Let's not bump version in PRs. Paul will bump it right before npm release.

File: package.json [hunk 3]
```diff
@@ -23,7 +23,8 @@
     "mocha": "^2.5.3",
     "chai": "^3.5.0",
     "eslint": "^3.0.1",
-    "eslint-config-brunch": "brunch/eslint-config-brunch"
+    "eslint-config-brunch": "brunch/eslint-config-brunch",
```
Comment (denysdovhan) [inline package.json]:
```
"eslint-config-brunch": "^1.0.0",
```

File: index.js (hunk 1 above)
Comment (denysdovhan) [inline index.js]: `Object.prototype.toString.call()` meh

File: test.js [hunk 4]
```diff
@@ -149,3 +149,27 @@ describe('Plugin Import Module', () => {
     });
   });
 });
+
+
+describe('Autprefix', () => {
```
Comment (denysdovhan) [inline test.js]: > Aut**o**prefix

or even better to give a more descriptive name e.g:

> Plugin as a function

File: index.js (hunk 1 above)
Comment (artisin) [inline index.js]: I feel you. I have a strong distaste for `typeof` because it has burned me in the past, but I made the requested change.

Comment (shvaikalesh) [inline index.js]: We don't support runtimes that have buggy `typeof`.

**Pull request merged**